src/Raven.Client/Http/RequestExecutor.cs

```diff
@@ -195,15 +228,20 @@ public void Execute<TResult>(RavenCommand<TResult> command, JsonOperationContext
 
         public async Task ExecuteAsync<TResult>(RavenCommand<TResult> command, JsonOperationContext context, CancellationToken token = default(CancellationToken))
         {
-            var choosenNode = ChooseNodeForRequest(command);
+            //run it only the first time we execute this function
+            if (_requiresTopologyUpdates && Interlocked.CompareExchange(ref _hasUpdatedTopologyAtLeastOnce, 1, 0) == 0)
```

Comment:
Bad, if we have concurrent calls, the first one will run this, the other will not wait